server/src/main/java/io/crate/execution/ddl/tables/TransportCreateTableAction.java

```diff
@@ -126,11 +129,15 @@ protected void masterOperation(final CreateTableRequest createTableRequest,
         CreateIndexRequest createIndexRequest = null;
         PutIndexTemplateRequest putIndexTemplateRequest = null;
         if (state.nodes().getMinNodeVersion().onOrAfter(Version.V_5_4_0)) {
+            Metadata.Builder metadataBuilder = Metadata.builder(state.metadata());
             if (createTableRequest.partitionedBy().isEmpty()) {
-                createIndexRequest = toCreateIndexRequest(createTableRequest);
+                createIndexRequest = toCreateIndexRequest(createTableRequest, metadataBuilder.columnOidSupplier());
             } else {
-                putIndexTemplateRequest = toPutIndexTemplateRequest(createTableRequest);
+                putIndexTemplateRequest = toPutIndexTemplateRequest(createTableRequest, metadataBuilder.columnOidSupplier());
             }
+            state = ClusterState.builder(state)
+                .metadata(metadataBuilder)
+                .build(); // Reflect OID advancing in the state before passing to services.
```

Comment:
I don't think it is safe to do the oid assignment here within `masterOperation`. It goes async afterwards before it submits a cluster state update task - that means the state can still change concurrently.

We have to do this in cluster state update tasks like https://github.com/crate/crate/blob/278ee3712d2fda38e8792cf4f89f749594484345/server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java#L259

Reply:
thx, done in https://github.com/crate/crate/pull/14142/commits/bb4b6701c728704fd6b5bd92d6194980c5a6e1a6

Propagating request to the relevant cluster state update tasks (index/template creation) so that they are able to build mapping with oid. 

Used a bit modified version of `TransportPutIndexTemplateAction/TransportCreateIndexAction masterOperation` to be able to pass mapping directly for BWC code.